public/app/features/auth-config/fields.tsx

```diff
@@ -620,6 +627,63 @@ export function fieldMap(provider: string): Record<string, FieldData> {
         'If enabled, Grafana will match the Hosted Domain retrieved from the Google ID Token against the Allowed Domains list specified by the user.',
       type: 'checkbox',
     },
+    serverDiscoveryUrl: {
+      label: 'Server discovery URL',
+      description:
+        'The .well-known/openid-configuration endpoint for your IdP. The info extracted from this URL will be used to populate the Auth URL, Token URL and API URL fields.',
+      type: 'custom',
+      content: (setValue, watch) => {
+        const modalIsOpen = watch('serverDiscoveryModal');
+        const onClose = () => setValue('serverDiscoveryModal', false);
+        const onSuccess = (data: ServerDiscoveryFormData) => {
+          fetch(data.url)
```

Comment:
I like the use of promises, but I received many comments from the frontend squad that the use of async/await is encouraged to be more consistent with the rest of the application.

Reply:
I modified the code to use async/await pattern instead of promises.